File: frontend/src/types/telegram.ts
// Telegram API types
// Must match backend/app/schemas/telegram.py

export interface TelegramUser {
  id: number
  username: string | null
  first_name: string | null
  last_name: string | null
  phone: string | null
}

export interface AuthStatusResponse {
  authenticated: boolean
  configured: boolean
  connected: boolean
  user: TelegramUser | null
}

export interface AuthStartRequest {
  phone: string
}

export interface AuthStartResponse {
  status: 'code_required'
  phone_code_hash: string
}

export interface AuthVerifyRequest {
  phone: string
  code: string
  phone_code_hash: string
  password?: string
}

export interface AuthVerifyResponse {
  status: 'authenticated' | '2fa_required'
}

export interface AuthLogoutResponse {
  status: 'logged_out'
}

export interface TelegramErrorResponse {
  error: string
  message: string
  retry_after: number | null
}

// Connection status for UI display
export type TelegramConnectionStatus =
  | 'disconnected'    // Not authenticated, click to set up
  | 'not_configured'  // API credentials not set
  | 'connecting'      // Auth in progress
  | 'connected'       // Authenticated with Telegram
  | 'unknown'         // API not responding

// Channel Resolution types
export interface ChannelResolveRequest {
  link: string
}

export interface ChannelResolveResponse {
  id: number | null
  title: string
  username: string | null
  type: 'channel' | 'supergroup' | 'group'
  members_count: number | null
  photo_url: string | null
  is_invite: boolean
  invite_hash: string | null
}

File: backend/app/schemas/telegram.py
"""Pydantic schemas for Telegram API."""

from __future__ import annotations

from pydantic import BaseModel, Field


# === Auth Request Schemas ===


class AuthStartRequest(BaseModel):
    """Request to start Telegram authentication."""

    phone: str = Field(
        ...,
        min_length=5,
        max_length=20,
        description="Phone number in international format (e.g., +1234567890)",
        examples=["+1234567890"],
    )


class AuthVerifyRequest(BaseModel):
    """Request to verify authentication code."""

    phone: str = Field(..., description="Phone number used in start request")
    code: str = Field(
        ...,
        min_length=1,
        max_length=10,
        description="Verification code received via SMS/Telegram",
    )
    phone_code_hash: str = Field(..., description="Hash returned from start request")
    password: str | None = Field(
        None, description="2FA password if required"
    )


# === Auth Response Schemas ===


class AuthStartResponse(BaseModel):
    """Response after starting authentication."""

    status: str = Field(..., description="Status: 'code_required'")
    phone_code_hash: str = Field(..., description="Hash to use in verify request")


class AuthVerifyResponse(BaseModel):
    """Response after verifying code."""

    status: str = Field(..., description="Status: 'authenticated' or '2fa_required'")


class TelegramUser(BaseModel):
    """Telegram user information."""

    id: int = Field(..., description="Telegram user ID")
    username: str | None = Field(None, description="Telegram username")
    first_name: str | None = Field(None, description="First name")
    last_name: str | None = Field(None, description="Last name")
    phone: str | None = Field(None, description="Phone number")


class AuthStatusResponse(BaseModel):
    """Response with current authentication status."""

    authenticated: bool = Field(..., description="Whether the session is authenticated")
    configured: bool = Field(..., description="Whether API credentials are configured")
    connected: bool = Field(..., description="Whether connected to Telegram")
    user: TelegramUser | None = Field(None, description="Current user if authenticated")


class AuthLogoutResponse(BaseModel):
    """Response after logout."""

    status: str = Field(default="logged_out", description="Status: 'logged_out'")


# === Error Response Schemas ===


class TelegramErrorResponse(BaseModel):
    """Error response for Telegram operations."""

    error: str = Field(..., description="Error code")
    message: str = Field(..., description="Human-readable error message")
    retry_after: int | None = Field(
        None, description="Seconds to wait before retrying (for rate limits)"
    )
